addon/-private/system/coerce-id.ts

```diff
@@ -1,10 +1,16 @@
+export type DirtyId =
+| null
+| undefined
+| string
+| number;
```

Comment:
I see a lot of named & exported interfaces in situations where we need these types _once_. We should decide whether this will cause needless clutter in e-d's internal API surface, compared to something like 
```ts
type Maybe<T> = T | null; // defined somewhere else

export default function coerceId(id?: Maybe<string|number>): Maybe<string>;
```

Reply:
I made a `Nullable<T>`, as `Maybe` made me feel like `undefined` would _maybe_ be included. :)